src/System.Reflection.Metadata/specs/Ecma-335-Issues.md

```diff
@@ -51,8 +51,80 @@ cycles in signatures, but see #2 below.
 Related issue:
 * https://github.com/dotnet/roslyn/issues/7970
 
+#### Proposed specification change:
 
-### 2.  `(CMOD_OPT | CMOD_REQ) <TypeSpec>)` is permitted in practice
+a) Rename section II.23.2.8 from "TypeDefOrRefOrSpecEncoded" to "TypeDefOrRefEncoded and TypeDefOrRefOrSpecEncoded"
+
+b) Replace
+
+*These items are compact ways to store a TypeDef, TypeRef, or TypeSpec token in a Signature (§II.23.2.12).*
+
+with 
+
+*TypeDefOrRefEncoded is a compact representation of either TypeDef or TypeRef token in a Signature (§II.23.2.12). TypeDefOrRefOrSpecEncoded is a compact representation of either TypeDef, TypeRef or TypeSpec token in a Signature.*
+
+Also correct 
+
+*The encoded version of this TypeRef token is made up as follows:*
+
+to
+
+*The compact representation of a TypeDef, TypeRef or TypeSpec token is made up as follows:*
+
+c) In section II.23.2.12 replace
+
+```
+Type ::=
+      ...
+      CLASS TypeDefOrRefOrSpecEncoded
+      ...
+      GENERICINST (CLASS | VALUETYPE) TypeDefOrRefOrSpecEncoded GenArgCount Type*
+      ...
+      VALUETYPE TypeDefOrRefOrSpecEncoded
+      ...
+```
+
+with
+
+```
+Type ::=
+      ...
+      (CLASS | VALUETYPE) TypeDefOrRefEncoded
+      GENERICINST (CLASS | VALUETYPE) TypeDefOrRefEncoded  GenArgCount Type+
+      ...
+```
+
+Note also the correction of ```Type*``` to ```Type+```. A generic type instantiation shall have at least one type argument.
+
+d) In section replace
+
+```
+TypeSpecBlob ::=
+      ...
+      GENERICINST (CLASS | VALUETYPE) TypeDefOrRefOrSpecEncoded GenArgCount Type Type*
+      ...
+```
+
+with
+
+```
+TypeSpecBlob ::=
+      ...
+      GENERICINST (CLASS | VALUETYPE) TypeDefOrRefEncoded GenArgCount Type+
+      ...
+```
+
+(```Type Type*``` is simplified to ```Type+```).
+
+#### Rationale of the proposal
+
+1) The proposal removes the possibility of representing the same type via two different encodings. This approach is consistent with II.23.2.16: "Short form signatures" where a short form of a primitive type is preferred over the corresponding long form.
```

Comment:
:bulb: Each item here can start with `1.` instead of `1)` to form a numbered list.